addon/types/model_export/model.py

```diff
@@ -172,13 +177,36 @@ def generate_qc(self):
             qc.write('$mostlyopaque')
             qc.write('\n')
 
+        output_x = 0
+        output_y = 0
+        output_z = 0
+        output_rotation = 0
+        output_scale = 1
+
+        if self.custom_transform_source == 'MANUAL':
+            output_x = self.origin_x
+            output_y = self.origin_y
+            output_z = self.origin_z
+            output_rotation = self.rotation
+            output_scale = self.scale
+        elif self.custom_transform_source == 'CUSTOM_OBJECT' and self.custom_transform_object_ref is not None:
+            output_x = (self.custom_transform_object_ref.matrix_world[0][3]) * -1
+            output_y = self.custom_transform_object_ref.matrix_world[1][3]
+            output_z = (self.custom_transform_object_ref.matrix_world[2][3]) * -1
```

Comment:
This is an interesting way of working with the matrix, but perhaps not the most readable; I suggest using [Matrix.decompose()](https://docs.blender.org/api/current/mathutils.html#mathutils.Matrix.decompose).
You can turn the rotation Quaternion into euler angles with [Quaternion.to_euler()](https://docs.blender.org/api/current/mathutils.html#mathutils.Quaternion.to_euler) (no need to set `euler_compat`, the default is good).

Reply:
I think i've done what you asked.
As for the matrix, it still works, but seemingly to get the offset to match in HLMV, all axes seem to have to be reversed?
Not sure if this is a bug with HLMV or if I am misinterpreting how `$origin` is working.

Reply:
![image](https://user-images.githubusercontent.com/87149601/126539765-4874639e-f281-45c7-a571-eec17b59ee45.png)

Reply:
I realise that my newest change isn't that much more readable. I'm just used to trying to reduce the amount of variables. I can add more for readability in that block.